### PR TITLE
Change production example to be an integration test

### PR DESCRIPTION
**Pull request by kfox1111**
The documentation is a better example then the test now. Separate the two.

Fixes https://github.com/spiffe/helm-charts-hardened/issues/217

Related to: https://github.com/spiffe/helm-charts-hardened/pull/206

### CODE REVIEW

**Review by marcofranssen — COMMENTED** [review]
Maybe we should decouple the examples from the integration `values.yaml`. For the purpose of an example it would be more transparant to find that values.yaml in the same folder. Maybe we should decouple that part and just go with the default values for an example and just only define the values required for that specific example.

**Comment by kfox1111** [comment]
> Maybe we should decouple the examples from the integration `values.yaml`. For the purpose of an example it would be more transparant to find that values.yaml in the same folder. Maybe we should decouple that part and just go with the default values for an example and just only define the values required for that specific example.

Hmm... What if we moved the run-tests.sh's of all of these tests into the tests/integration subdir instead of examples then link back to the values in examples?  Then the users would be less likely to look at the run-tests.sh and get confused by it while still allowing us to test production configurations of the example values?

**Comment by kfox1111** [comment]
@marcofranssen I tried to take a stab at incorporating your feedback. Please rereview when you get a chance.

File: .github/tests/common.sh [hunk 1]
```diff
@@ -70,3 +70,23 @@ $(helm ls -A | sed 's/\t/ | /g' | sed 's/^/| /' | sed 's/$/ |/' | sed '/^| NAME.
...
+global:
+  spire:
+    recommendations:
+      enabled: true
+    clusterName: production
+    trustDomain: production.other
+    caSubject:
+      country: US
+      organization: Production
+      commonName: production.other
```
Comment (faisal-memon) [inline .github/tests/common.sh:86]: Why have the values like this rather than in a file?

**Review by faisal-memon — APPROVED** [review]

**Pull request merged**